File: frontend/src/services/StudentService.ts
import axios from "axios";
import { Student } from "../models/Student";

const API_URL = "http://localhost:8080/api/students";

export const getAllStudents = () =>
  axios.get<Student[]>(API_URL);

export const getStudentById = (id: number) =>
  axios.get<Student>(`${API_URL}/${id}`);

export const createStudent = (student: Student) =>
  axios.post<Student>(API_URL, student);

export const updateStudent = (id: number, student: Student) =>
  axios.put<Student>(`${API_URL}/${id}`, student);

export const deleteStudent = (id: number) =>
  axios.delete(`${API_URL}/${id}`);

File: backend/src/main/java/com/example/student/service/StudentService.java
package com.example.student.service;

import com.example.student.model.Student;

import java.util.List;

public interface StudentService {

    List<Student> getAllStudents();

    Student getStudentById(Long id);

    Student createStudent(Student student);

    Student updateStudent(Long id, Student student);

    Student partialUpdateStudent(Long id, Student student);

    void deleteStudent(Long id);
}
